### PR TITLE
DTSRD-1992- Add a check for zero byte characters durng file upload

### PR DESCRIPTION
**Pull request by SabinaHMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by mikebrownccd — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/CourtVenueProcessor.java [hunk 1]
```diff
@@ -87,14 +98,42 @@ public void process(Exchange exchange) throws Exception {
...
         exchange.getMessage().setBody(filteredCourtVenues);
+
+    }
+
+
+    private void processExceptionRecords(Exchange exchange,
+                                         List<CourtVenue> courtVenuesList) {
+
+        List<Pair<String, Long>> zeroByteCharacterRecords = new ArrayList<>();
+        courtVenuesList.forEach(courtVenue -> dataQualityCheckConfiguration.zeroByteCharacters
```
Comment (lukasz-wolski) [inline src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/CourtVenueProcessor.java:117]: public field dereference, please consider defining method for zeroByteCharacters and accessing via method

File: src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/BuildingLocationProcessor.java [hunk 2]
```diff
@@ -75,12 +88,41 @@ public void process(Exchange exchange) throws Exception {
...
 
         exchange.getMessage().setBody(validatedBuildingLocations);
+
+    }
+
+
+    private void processExceptionRecords(Exchange exchange,
+                                         List<BuildingLocation> buildingLocationsList) {
+        List<Pair<String, Long>> zeroByteCharacterRecords = new ArrayList<>();
+        buildingLocationsList.forEach(buildingLoc -> dataQualityCheckConfiguration.zeroByteCharacters
```
Comment (lukasz-wolski) [inline src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/BuildingLocationProcessor.java:106]: public field dereference, please consider defining method for zeroByteCharacters and accessing via method

**Review by lukasz-wolski — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/BuildingLocationProcessor.java (hunk 2 above)
Comment (SabinaHMCTS) [reply src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/BuildingLocationProcessor.java:118]: defined zerobyteCharacterCheck moved zerobyte check to method and added null check

Comment (lukasz-wolski) [reply src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/BuildingLocationProcessor.java:118]: zeroBytesCharacters is still being accessed outside of object directly. please introduce 'getter' method and access via method

File: src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/CourtVenueProcessor.java (hunk 1 above)
Comment (lukasz-wolski) [reply src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/CourtVenueProcessor.java:126]: zeroBytesCharacters is still being accessed outside of object directly. please introduce 'getter' method and access via method

**Review by lukasz-wolski — CHANGES_REQUESTED** [review]
please see comments inlined

File: src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/CourtVenueProcessor.java (hunk 1 above)
Comment (SabinaHMCTS) [reply src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/CourtVenueProcessor.java]: added getter setters

File: src/main/java/uk/gov/hmcts/reform/locationrefdata/configuration/DataQualityCheckConfiguration.java [hunk 3]
```diff
@@ -9,5 +9,13 @@
...
-    public List<String> zeroByteCharacters;
+    private List<String> zeroByteCharacters;
+
+    public List<String> getZeroByteCharacters() {
+        return zeroByteCharacters;
+    }
+
+    public void setZeroByteCharacters(List<String> zeroByteCharacters) {
+        this.zeroByteCharacters = zeroByteCharacters;
+    }
```
Comment (lukasz-wolski) [inline src/main/java/uk/gov/hmcts/reform/locationrefdata/configuration/DataQualityCheckConfiguration.java:20]: a setter is not needed, let spring inject value into private field (setter method would also have to be unit tested if not removed). please consider removing setter method.

**Review by lukasz-wolski — CHANGES_REQUESTED** [review]
please see comment above

**Review by lukasz-wolski — CHANGES_REQUESTED** [review]
see comment inlined

File: src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/BuildingLocationProcessor.java (hunk 2 above)
Comment (SabinaHMCTS) [reply src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/BuildingLocationProcessor.java]: added getters and setters

**Review by lukasz-wolski — APPROVED** [review]

**Review by mikebrownccd — APPROVED** [review]
LGTM

**Pull request merged**